src/libsystemd/sd-cloud-provider/sd-cloud-provider.c

```diff
@@ -0,0 +1,190 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <errno.h>
+#include <poll.h>
+#include <sys/inotify.h>
+
+#include "sd-cloud-provider.h"
+
+#include "alloc-util.h"
+#include "env-file.h"
+#include "fd-util.h"
+#include "fs-util.h"
+#include "macro.h"
+#include "parse-util.h"
+#include "stdio-util.h"
+#include "string-util.h"
+#include "strv.h"
+#include "util.h"
+
+static int cloud_provider_link_get_string(int ifindex, const char *field, char **ret) {
+        char path[STRLEN("/run/systemd/cloud-provider/netif/links/") + DECIMAL_STR_MAX(ifindex) + 1];
```

Comment:
Any specific reason to use file i/o, rather than dbus, as the transport between providerd and clients?

Reply:
We will add dbus and varlink link and dbus in further enhancements.